viewing_party/party.py

```diff
@@ -0,0 +1,199 @@
+		""" In each iteration movie takes on the value of each element in the movie_list.
+			Ex. Iteration 0: 
+			    movie = {"genre": "Horror"}
+			    genre = "Horror"
+			    Iteration 1:
+			    movie = {"genre": "Action"}
+				genre = "Action"
+				ETC
+		"""
+		if genre in count_genres:
+			""" Assume "Horror" is an existing key in count_genres
+			    count_genres = {
+					"Horror": 1,
+					"Action": 1
+			    } 
+				Ex. Iteration 3
+					movie = {"genre": "Horror"}
+			    	genre = "Horror"
+			"""
+			count_genres[genre] += 1
+		else:
+			count_genres[genre] = 1
+		if count_genres[genre] > highest_count:
+			most_watched_genre = genre
+			highest_count = count_genres[genre]
+	return most_watched_genre
+
+def get_unique_watched(user_data):
+    """ user_data dict with watched list of movies dictionaries and friends
+    user_data = {
+        watched[movies{title:}],
+        friends[friend{watched[movies{title:}]}]                  
+    }
+    """
+    """
+movies that user has watched
+movies that friend has watched
+which movies the user has watched but none friends watched  movies user watch != movies friends watched 
+return unique[movies{}]
+    """
+    unique_movies = []
+
+    movie_list = user_data["watched"]
+    friends_list = user_data["friends"]
+
+    for movie in movie_list:
+        title = movie["title"]
+        unique = True
+        for friend in friends_list:
+            friend_movies_list = friend["watched"]
+            for friend_movie in friend_movies_list:
+                if friend_movie["title"] == title:
+                    unique = False
+                    break
+            if not unique:
+                break
+        if unique:
+            unique_movies.append(movie)
+
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+    unique_movies = []
+    movie_list = user_data["watched"]
+    friends_list = user_data["friends"]
+
+    for friend in friends_list:
+        friend_movies_list = friend["watched"]
+        for friend_movie in friend_movies_list:
+            title = friend_movie["title"]
+            unique = True    
+            for movie in movie_list:
+                if movie["title"] == title:
+                    unique = False
+                    break
+            if unique and friend_movie not in unique_movies:
+                unique_movies.append(friend_movie)
+
+    return unique_movies
+
+def get_available_recs(user_data):
+    """user_data
+    {    subscriptions["name, name , name"]}
+    """
+    unique_movies = []
+    movie_list = user_data["watched"]
+    friends_list = user_data["friends"]
+
+    for friend in friends_list:
+        friend_movies_list = friend["watched"]
+        for friend_movie in friend_movies_list:
+            title = friend_movie["title"]
+            unique = True    
+            for movie in movie_list:
+                if movie["title"] == title:
+                    unique = False
+                    break
```

Comment:
Notice how similar this code is to some of the previous functions. How could we simplify this code if we used a function that gave us the movies watched by the user's friends, but not the user?